lib/Backends/DeviceManagers.cpp

```diff
@@ -0,0 +1,71 @@
+/**
```

Comment:
@nickgg To add a new backend we now have to add code in two places. One way to simplify this and make it one place would be to make the device manager a member of the backend. Something like Backend::createDeviceManager. 

This is just an idea. I am not sure if this is a problem that we need to solve.

Reply:
Yeah I definitely think that would be more convenient, my concern here would be coupling between the Backend and the DeviceManager.

In an ideal world I think the compile time and runtime parts of glow would be distinct, connected by the Provisioner and the CompiledFunction.